http-client/src/main/java/no/nav/familie/http/interceptor/BearerTokenClientInterceptor.kt

```diff
@@ -19,9 +19,39 @@ class BearerTokenClientInterceptor(private val oAuth2AccessTokenService: OAuth2A
                                    private val clientConfigurationProperties: ClientConfigurationProperties) :
```

Comment:
Ble kanskje litt mye med 3 veldig like... Alternativt å bruke abstract med en provider for OAuth2GrantType
```
abstract class AbstractBearerTokenClientInterceptor(private val oAuth2AccessTokenService: OAuth2AccessTokenService,
                                                    private val clientConfigurationProperties: ClientConfigurationProperties,
                                                    private val grantType: () -> OAuth2GrantType) : ClientHttpRequestInterceptor {

    override fun intercept(request: HttpRequest, body: ByteArray, execution: ClientHttpRequestExecution): ClientHttpResponse {
        request.headers.setBearerAuth(genererAccessToken(request,
                                                         clientConfigurationProperties,
                                                         oAuth2AccessTokenService,
                                                         grantType.invoke()))
        return execution.execute(request, body)
    }
}
```

Reply:
Ikke krise tenker jeg, begrenset hvor ofte man endrer og legger til nye av denne uansett.